clang/lib/CodeGen/CoverageMappingGen.cpp

```diff
@@ -722,6 +713,9 @@ struct MCDCCoverageBuilder {
       return I->second;
   }
 
+  /// Return the LHS Decision ([0,0] if not set).
+  const DecisionIDPair back() const { return DecisionStack.back(); }
```

Comment:
I think this may be byref as far as the callee handles byvalue.

Reply:
Ah, you're right.

Reply:
On second thought, could you elaborate on your concern here? I apologize; I know DecisionStack.back() returns a reference, but this ought to now copy that value, right?

Reply:
This may return byref since `DecisionStack.back()` returns byref. I think this is consistent.
The callee (visitBin) may take an option to copy values instead of holding the ref.

I guess the change doesn't have any big impacts for generated code. I suggest just for the consistency.

Reply:
Ah, ok I think I misunderstood.  You're suggesting that the fact taht DecisionStack.back() returns byref needs to be reflected by the return type of this function. In other words, restore the return type to the way it was:

`const DecisionIDPair &back() const { return DecisionStack.back(); }`

I did change the visitBin* functions to no longer hold a reference.